### PR TITLE
Optionally send /sync requests to a separate process than synapse

### PR DESCRIPTION
**Pull request by NegativeMjark**

### CODE REVIEW

File: jenkins.sh [hunk 1]
```diff
@@ -42,7 +39,8 @@ virtualenv $WORKSPACE/.venv
 PYTHON=$WORKSPACE/.venv/bin/python
 PIP=$WORKSPACE/.venv/bin/pip
 (cd synapse ; $PYTHON synapse/python_dependencies.py | xargs -n1 $PIP install)
-
+$PIP install psycopg2
+$PIP install lxml
```
Comment (kegsay) [inline jenkins.sh:43]: Why? Shouldn't this be in `python_dependencies.py`?

File: src/github.com/matrix-org/dendron/dendron/main.go [hunk 2]
```diff
@@ -197,6 +224,19 @@ func main() {
...
 	mux.Handle("/_matrix/client/versions", versionsFunc)
+
+	if synchrotronURL != nil {
+		synchrotronReverseProxy := proxy.MeasureByPath(
+			proxyMetrics,
+			httputil.NewSingleHostReverseProxy(synchrotronURL).ServeHTTP,
+		)
+		synchrotronFunc := prometheus.InstrumentHandler(
+			"synchrotron", synchrotronReverseProxy,
+		)
```
Comment (kegsay) [inline src/github.com/matrix-org/dendron/dendron/main.go:235]: A line explaining what `synchrotronReverseProxy` and `synchrotronFunc` are doing would be nice.

**Comment by kegsay** [comment]
LGTM aside from comments

File: jenkins.sh (hunk 1 above)
Comment (NegativeMjark) [inline jenkins.sh:43]: We special case them in synapse for reasons. This is copied from the synapse jenkins.sh

Comment (kegsay) [inline jenkins.sh:43]: > This is copied from the synapse jenkins.sh

Comment this then please so if the Synapse `jenkins.sh` changes and we don't need to do that anymore, we know we can safely nuke this bit as well.

**Pull request merged**